src/Config/Utilities/FileUtilities.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using System.Security.Cryptography;
+
+namespace Azure.DataApiBuilder.Config.Utilities;
+
+internal class FileUtilities
+{
+    /// <summary>
+    /// Computes the SHA256 hash for the input data (file contents) at the given path.
+    /// Includes an exponential back-off retry mechanism to accommodate
+    /// circumstances where the file may be in use by another process.
+    /// "The hash is used as a unique value of fixed size representing a large amount of data.
+    /// Hashes of two sets of data should match if and only if the corresponding data also matches.
+    /// Small changes to the data result in large unpredictable changes in the hash."
+    /// This utility function code is based on and modified from the example provided in the
+    /// .NET ChangeToken documentation.
+    /// </summary>
+    /// <param name="fileSystem">File system abstraction.</param>
+    /// <param name="filePath">Abosolute file path or relative file path.</param>
+    /// <returns>32 byte message digest.</returns>
+    /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.sha256?view=net-8.0#remarks"/>
+    /// <seealso cref="https://learn.microsoft.com/en-us/aspnet/core/fundamentals/change-tokens?view=aspnetcore-8.0#:~:text=exponential%20back%2Doff.-,Utilities/Utilities.cs%3A,-C%23"/>
+    /// <exception cref="FileNotFoundException">https://learn.microsoft.com/en-us/dotnet/api/system.io.file.openread?view=net-8.0#exceptions</exception>
+    public static byte[] ComputeHash(IFileSystem fileSystem, string filePath)
+    {
+        // Exponential back-off retry mechanism.
+        int runCount = 1;
+
+        // Arbitrary limit of 3 attempts to load the file.
+        // Maximum 8 seconds (2^3) of wait time due to retries.
+        while (runCount < 4)
```

Comment:
nit: prefer to use a named variable like "runLimit" in place of magic number '4'

Reply:
updated with:
```csharp
    /// <summary>
    /// Limit the number of retries when failures occur.
    /// </summary>
    private static readonly int _runLimit = 3;

    /// <summary>
    /// Base of the exponential retry back-off.
    /// -> base ^ runCount
    /// e.g. 2^1 , 2^2, 2^3, etc.
    /// </summary>
    private static readonly int _exponentialRetryBase = 2;
```